src/analysis/topology/conservedMoieties/identifyConservedMoieties.m

```diff
@@ -119,67 +108,110 @@
-                [~,row2] = sort(mets2);
-                row2row = sortrows([row2 row1],1); % mets2 = mets1(row2row)
-
-                [~,col1] = sort(rxns1);
-                [~,col2] = sort(rxns2);
-                col2col = sortrows([col2 col1],1); % rxns2 = rxns1(col2col)
-
-                if all(all(mgraph2 == mgraph1(row2row,col2col)))
-                    atoms2instances(comp2) = rowidx(row2row); % map atoms to moieties
-                else
-                    warning('atom graphs not isomorphic'); % Hopefully we never get here. Complicates things.
-                end
+                continue;
+            end
+            
+            % Attempt to match component graphs by sorting
+            [mgraph1,perm] = sortrows([mets1 mgraph1]);
+            mets1 = mgraph1(:,1);
+            mgraph1 = mgraph1(:,2:end);
+            comp1 = comp1(perm);
+            rowidx = rowidx(perm);
+            
+            [mgraph1,perm] = sortrows([rxns1 mgraph1']);
+            rxns1 = mgraph1(:,1);
+            mgraph1 = mgraph1(:,2:end)';
+            trans1 = trans1(perm);
+            
+            [mgraph2,perm] = sortrows([mets2 mgraph2]);
+            mets2 = mgraph2(:,1);
+            mgraph2 = mgraph2(:,2:end);
+            comp2 = comp2(perm);
+            
+            [mgraph2,perm] = sortrows([rxns2 mgraph2']);
+            rxns2 = mgraph2(:,1);
+            mgraph2 = mgraph2(:,2:end)';
+            trans2 = trans2(perm);
+            
+            if (all(mets2 == mets1) && all(rxns2 == rxns1)) && all(all(mgraph2 == mgraph1))
+                atoms2instances(comp2) = rowidx; % map atoms to moieties
+                continue;
+            end
+            
+            % use Matlab's built in isomorphism algorithm (last resort because it's slow, should only need it in rare cases)
+            [g1,o2n1,n2o1] = unique(mgraph1','rows','stable'); % Matlab graphs do not support replicate edges
+            g1 = g1';
+            urxns1 = rxns1(o2n1);
+            utrans1 = trans1(o2n1);
+            [h1,~] = find(g1 < 0);
+            [t1,~] = find(g1 >0);
+            edges1 = table([h1 t1],urxns1,utrans1,'VariableNames',{'EndNodes' 'Rxn' 'Transition'});
+            nodes1 = table(mets1,comp1,'VariableNames',{'Met' 'Atom'});
+            d1 = digraph(edges1,nodes1);
+            
+            [g2,o2n2,n2o2] = unique(mgraph2','rows','stable');
+            g2 = g2';
+            urxns2 = rxns2(o2n2);
+            utrans2 = trans2(o2n2);
+            [h2,~] = find(g2 < 0);
+            [t2,~] = find(g2 >0);
+            edges2 = table([h2 t2],urxns2,utrans2,'VariableNames',{'EndNodes' 'Rxn' 'Transition'});
+            nodes2 = table(mets2,comp2,'VariableNames',{'Met' 'Atom'});
+            d2 = digraph(edges2,nodes2);
+            
+            % find isomorphism that conserves metabolite and reaction
+            % attributes of nodes and edges
+            p = isomorphism(d1,d2,'NodeVariables','Met','EdgeVariables','Rxn');
```

Comment:
Similar comment here: `isomorphiosm` was only introduced in 2016b and won't work on previous versions.